src/codeu/chat/client/simplegui/UserPanel.java

```diff
@@ -171,8 +171,12 @@ public void actionPerformed(ActionEvent e) {
             UserPanel.this, "Enter user name:", "Add User", JOptionPane.PLAIN_MESSAGE,
             null, null, "");
         if (s != null && s.length() > 0) {
-          clientContext.user.addUser(s);
-          UserPanel.this.getAllUsers(listModel);
+          if(clientContext.user.addUser(s)==true){
+          	UserPanel.this.getAllUsers(listModel);
+          	}
+          else{
+          	JOptionPane.showMessageDialog(UserPanel.this, "This username is already in use.", "Error", JOptionPane.ERROR_MESSAGE);
```

Comment:
Love that you are showing the user a dialog rather than just logging the error on the server side!